File: plotHawkes.m
function plotHawkes(tev,lambda_0,alpha_0,w,T, res)

tvec=0:max(T)/res:max(T);
mu_t= (exp((alpha_0-w).*tvec)+w*(alpha_0-w)^-1*(exp((alpha_0-w).*tvec)-1))*lambda_0;
plot(tvec, mu_t, 'b-','linewidth',1.5);


colorLambda={'r--','k--','g--','m--','c--'};
colorEv={'r+','k+','g+','m+','c+'};
for i=1:length(tev)
    n=0;
    lambda_t=zeros(1,length(tvec));
    for t=tvec
        n=n+1;
        lambda_t(n)= lambda_0+alpha_0*sum(exp(-w*(t-tev{i}(tev{i}<t))));
    end
    hold on, plot(tvec, lambda_t, colorLambda{mod(i,length(colorEv))+1}, tev{i},zeros(1,length(tev{i})), colorEv{mod(i,length(colorEv))+1});
end
legend('E_H[\lambda(t)]','\lambda(t)', 'events')

File: recurrent_events/plotHawkes.py
import matplotlib.pyplot as plt
import numpy as np

def plotHawkes(tev, l_0, alpha_0, w, T, res):
    tvec = np.arange(0, T, step=T / res)

    mu_t = (np.exp((alpha_0 - w) * tvec) + w * (1.0 / (alpha_0 - w)) *
            (np.exp((alpha_0 - w) * tvec) - 1)) * l_0

    plt.plot(tvec, mu_t, 'b-', linewidth=1.5)

    colorLambda = ['r--', 'k--', 'g--', 'm--', 'c--']
    colorEv = ['r+', 'k+', 'g+', 'm+', 'c+']

    for i in range(len(tev)):
        n = -1
        l_t = np.zeros(len(tvec))

        for t in tvec:
            n += 1
            l_t[n] = l_0 + alpha_0 * np.sum(np.exp(-w * (t - tev[i][tev[i] < t])))

        plt.plot(tvec, l_t, colorLambda[i % len(colorLambda)])
        plt.plot(tev[i], np.zeros(len(tev[i])), colorEv[i % len(colorEv)])


    plt.ylim([-0.1, None]) # To make sure that the event markers show.
    # plt.legend

